src/python/pants/backend/go/util_rules/first_party_pkg.py

```diff
@@ -134,15 +149,65 @@ async def compute_first_party_package_info(
         Get(FirstPartyPkgImportPath, FirstPartyPkgImportPathRequest(request.address)),
         Get(GoModInfo, GoModInfoRequest(owning_go_mod.address)),
     )
+    target = wrapped_target.target
+
+    # Find resource targets.
+    # TODO: This uses ExplicitlyProvidedDependencies to avoid triggering dependency inference which
+    # causes a rule cycle back to this rule as dependency inference tries to analyze dependent
+    # `go_package` targets. There really needs to be a way to limit `DependenciesRequest` to just
+    # targets meeting certain criteria, which would be resource targets in this case.
+    explicit_deps = await Get(
+        ExplicitlyProvidedDependencies, DependenciesRequest(target[Dependencies])
+    )
+    explicit_deps_targets = await Get(Targets, Addresses(explicit_deps.includes))
+    pkg_relative_resource_targets = [
+        tgt
+        for tgt in explicit_deps_targets
+        # Note: target expansion is not ocurring?
+        if tgt.has_field(ResourceSourceField) or tgt.has_field(ResourcesGeneratingSourcesField)
+        # TODO: Currently limited to resources directly under package in source tree. Allow any resource and
+        # have paths be relative to the resources' spec_path.
+        and tgt.address.spec_path.startswith(request.address.spec_path)
+    ]
 
     pkg_sources = await Get(
         HydratedSources,
-        HydrateSourcesRequest(wrapped_target.target[GoPackageSourcesField]),
+        HydrateSourcesRequest(target[GoPackageSourcesField]),
+    )
+
+    resources_sources = await MultiGet(
+        Get(
+            HydratedSources,
+            HydrateSourcesRequest(
+                tgt[SourcesField],
+                for_sources_types=(
+                    ResourceSourceField,
+                    ResourcesGeneratingSourcesField,
+                ),
+                enable_codegen=True,
+            ),
+        )
+        for tgt in pkg_relative_resource_targets
+    )
+
+    original_resources_digest = await Get(
+        Digest, MergeDigests([src.snapshot.digest for src in resources_sources])
+    )
+    stripped_resources_digest = await Get(
+        Digest, RemovePrefix(original_resources_digest, request.address.spec_path)
     )
+    resources_digest = await Get(Digest, AddPrefix(stripped_resources_digest, "__resources__"))
```

Comment:
This code puts the resources into a separate directory so that the analysis code doesn't consider source files for the wildcards. The analysis code is passed the `__resources__` path so it knows where the resources to consider are.